lib/Arch/Runtime/HyperCall.cpp

```diff
@@ -129,6 +129,14 @@ Memory *__remill_sync_hyper_call(State &state, Memory *mem,
       mem = __remill_x86_set_segment_gs(mem);
       break;
 
+    case SyncHyperCall::kX86SysEnter: asm volatile("sysenter" : :); break;
```

Comment:
I think these syscall-related inline asm blocks should take in the dwords of the main 7 or 8 gprs by value, e.g. rax, rbx, ..., even rsp, and output rax on 32-bit, and on 64 bit, they should take in even more, maybe even all general-purpose regs.

Reply:
Just for my own understanding, is that because the argument registers for `syscall` are OS dependent?

I chose the registers based on the table here, but I realise it says Linux specifically: https://www.cs.uaf.edu/2017/fall/cs301/lecture/11_17_syscall.html

Reply:
There is an ABI, that is for sure, but the semantics don't actually know the target OS / ABI, so I was erring on the side of "copy everything in." The 32- vs. 64-bit distinction is there because in 32-bit bitcode, you can't pass a 64-bit value in. I suppose you could use something like `"a"(state.gpr.aword)`, and the `aword` macro will expand to one of `dword` or `qword`.

Reply:
4f001ec